sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -661,6 +664,15 @@ private object CosmosReadConfig {
       "entry."
   )
 
+  private val RuntimeFilteringEnabled = CosmosConfigEntry[Boolean](
+      key = CosmosConfigNames.ReadRuntimeFilteringEnabled,
+      mandatory = false,
+      defaultValue = Some(false),
+      parseFromStringFunction = runtimeFilteringEnabled => runtimeFilteringEnabled.toBoolean,
+      helpMessage = "When enabled, will try to optimize the read using readMany vs query when spark dynamic partition pruning happens and there are runtime filters " +
+          "being pushed down. If for the customer's container, id is the also the partitionKey, this optimization will kick in automatically when applicable. Otherwise, " +
```

Comment:
nit: comment is not true, only consider _itemIdentity for now. Id optimization will come in a different PR